tests/dummy/app/pods/demo/controller.js

```diff
@@ -21,6 +22,31 @@ export default Controller.extend({
       clearDuration: 1000
     })
   },
+  init () {
+    this._super(...arguments)
+    let customRouteObject = Ember.Object.extend({
```

Comment:
This is just there as an example to illustrate adding routes dynamically, and updating a given route when certain properties are recomputed (ex. query params). 
Doesn't have to happen on the init hook.